otlp/collector/profiles/go.mod

```diff
@@ -0,0 +1,23 @@
+module go.opentelemetry.io/proto/otlp/collector/profiles
```

Comment:
We should add the Go modules deeper with `v1development` so that we can later deprecate them

Related comment https://github.com/open-telemetry/opentelemetry-proto-go/pull/358#issuecomment-3095544237

CC @florianl